liquidprompt

```diff
@@ -4081,22 +4086,27 @@ _lp_default_theme_activate() {
 }
 
 _lp_default_theme_directory() {
-    # LP_PERM: shows a ":"
-    # - colored in green if user has write permission on the current dir
-    # - colored in red if not
-    # - can set another symbol with LP_MARK_PERM
-    if (( LP_ENABLE_PERM )); then
-        if [[ -w "${PWD}" ]]; then
-            LP_PERM="${LP_COLOR_WRITE}${LP_MARK_PERM}${NO_COL}"
-        else
-            LP_PERM="${LP_COLOR_NOWRITE}${LP_MARK_PERM}${NO_COL}"
+    if ((LP_ENABLE_PATH)); then
+        # LP_PERM: shows a ":"
+        # - colored in green if user has write permission on the current dir
+        # - colored in red if not
+        # - can set another symbol with LP_MARK_PERM
+        if (( LP_ENABLE_PERM )); then
+            if [[ -w "${PWD}" ]]; then
+                LP_PERM="${LP_COLOR_WRITE}${LP_MARK_PERM}${NO_COL}"
+            else
+                LP_PERM="${LP_COLOR_NOWRITE}${LP_MARK_PERM}${NO_COL}"
+            fi
         fi
-    fi
 
-    local lp_path_format
-    _lp_path_format "$LP_COLOR_PATH" "$LP_COLOR_PATH_LAST_DIR" "$LP_COLOR_PATH_VCS_ROOT" "$LP_COLOR_PATH_SHORTENED" "/" "$LP_COLOR_PATH_SEPARATOR"
+        local lp_path_format
+        _lp_path_format "$LP_COLOR_PATH" "$LP_COLOR_PATH_LAST_DIR" "$LP_COLOR_PATH_VCS_ROOT" "$LP_COLOR_PATH_SHORTENED" "/" "$LP_COLOR_PATH_SEPARATOR"
```

Comment:
I think the check for `LP_ENABLE_PATH` really should be part of `_lp_path_format()`. While it's fine to keep the check at the top of this function, so far the API has been to do config checking inside the data functions.

Reply:
Good catch.